vscode-extension/src/utils.ts

```diff
@@ -0,0 +1,64 @@
+import * as path from "path";
+import * as fs from "fs/promises";
+
+async function exists(file: string): Promise<boolean> {
+  return fs
+    .stat(file)
+    .then(() => true)
+    .catch(() => false);
+}
+
+function getBinaryPathRelativeToPackageJson() {
+  let binaryPathRelativeToPackageJson;
+  if (process.platform === "darwin" && process.arch === "x64") {
+    binaryPathRelativeToPackageJson = path.join("macos-x64", "relay");
+  } else if (process.platform === "darwin" && process.arch === "arm64") {
+    binaryPathRelativeToPackageJson = path.join("macos-arm64", "relay");
+  } else if (process.platform === "linux" && process.arch === "x64") {
+    binaryPathRelativeToPackageJson = path.join("linux-x64", "relay");
+  } else if (process.platform === "win32" && process.arch === "x64") {
+    binaryPathRelativeToPackageJson = path.join("win-x64", "relay.exe");
+  } else {
+    binaryPathRelativeToPackageJson = null;
+  }
```

Comment:
Maybe add a comment that this mirrors what we're doing in [`relay-compiler/index.js`](https://github.com/facebook/relay/blob/14550c8bd80e53369852db66d7681f13d6dd5f84/packages/relay-compiler/index.js)? If we make a change in one place we probably want to make it in both.

Maybe also add a comment in that file pointing back here?